api/prd/location/index.ts

```diff
@@ -46,18 +46,26 @@ export async function getLocations(req: EnhancedRequest, res: Response, next: Ne
  */
 export async function getLocationById(req: EnhancedRequest, res: Response, next: NextFunction) {
   const epimmsID = req.query.epimms_id;
+  const serviceCode = !!req.query.serviceCode ? req.query.serviceCode as string : null;
+  delete req.query.serviceCode;
+  const courtTypeIdsArray: string[] = getCourtTypeIdsByServices([serviceCode]);
   const markupPath: string = `${url}/refdata/location/court-venues?epimms_id=${epimmsID}`;
   try {
     const { status, data }: { status: number, data: LocationModel[] } = await handleGet(markupPath, req, next);
-    const identicalLocationByEpimmsId = getIdenticalLocationByEpimmsId(data);
+    const courtLocations = serviceCode && courtTypeIdsArray?.length > 0 ? getLocationsByCourtType(data, courtTypeIdsArray) : data;
+    const identicalLocationByEpimmsId = getIdenticalLocationByEpimmsId(courtLocations);
     res.status(status).send(identicalLocationByEpimmsId);
   } catch (error) {
     next(error);
   }
 }
 
-function getIdenticalLocationByEpimmsId(data: LocationModel[]) {
+function getIdenticalLocationByEpimmsId(data: LocationModel[]): LocationByEPIMMSModel[] {
```

Comment:
can we be more consistent with capitlisation please. I suggest locationsByEpimmsModel

Reply:
Have changed all instances to be the suggested name